### PR TITLE
Yura/cpa26 add poem to bookmark

### PR DESCRIPTION
**Pull request by yuriy1901**
Made bookmarks logic on event click, page of bookmarks with list of poems, login/logout logic with bookmarks
Did fix with img sizes in the poems list - CPA96_Fix Poem.Header 
Add some new logic to basic axios request - now you can dispatch extra data from component - not only from response from API

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -37,6 +41,7 @@ function App() {
 			<Routes>
 				<Route path="/" element={<HomePage />} />
 				<Route path={ROUTES.ROUTE_AUTHOR_PAGE} element={<AuthorPage />} />
+				<Route path={ROUTES.ROUTE_BOOKMARKS_PAGE} element={<BookMarksPage />} />
```
Comment (RudeAnna) [inline src/App.js:44]: Typo. The name of the component should be <BookmarksPage />
"A book mark" means a bit different from "a bookmark"

File: src/components/Poem/Poem.js [hunk 2]
```diff
@@ -164,10 +166,10 @@ Poem.ReadMoreButton = function PoemReadMoreButton({ ...restProps }) {
 };
 
 Poem.TagsList = function PoemTagslist({ poem, children, ...restProps }) {
-	const arrayToShow = Array.isArray(poem?.tags) ? poem.tags : poem.tags.split(',');
+	const arrayToShow = Array.isArray(poem?.tags) ? poem.tags : poem.tags?.split(',');
```
Comment (RudeAnna) [inline src/components/Poem/Poem.js:169]: maybe it's better to name what kind of information is in the array arrayToShow, like poemTagsArray, so it's more clear

File: src/components/Poem/Poem.js [hunk 3]
```diff
@@ -182,3 +184,7 @@ Poem.TagsList = function PoemTagslist({ poem, children, ...restProps }) {
 Poem.VoteBlock = function PoemVoteBlock({ poem }) {
 	return <PoemVotes poem={poem} />;
 };
+
+Poem.BookmarBlock = function PoemBookmarkBlock({ poem }) {
```
Comment (RudeAnna) [inline src/components/Poem/Poem.js:188]: typo BookmarkBlock

File: src/pages/BookMarksPage/BookMarksPage.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import {getBookMarkedPoemsAsyncAction} from "../../redux/poems/poemsAsyncActions";
+import PoemsList from "../../components/PoemsList/PoemsList";
+
+export default function BookMarksPage() {
+
+    const dispatch = useDispatch();
+
+    useEffect(() => {
+        window.scrollTo(0, 0);
+        dispatch(getBookMarkedPoemsAsyncAction());
```
Comment (RudeAnna) [inline src/pages/BookMarksPage/BookMarksPage.js:12]: SAP Bookmarked

**Review by MuscularSloth — APPROVED** [review]

File: src/components/Poem/Poem.js (hunk 3 above)
Comment (yuriy1901) [reply src/components/Poem/Poem.js:188]: I think PoemBookmarkBlock will be better then BookmarkBlock, becouse this block is always in poem, and it is in Poem component

File: src/components/Poem/Poem.js (hunk 2 above)
Comment (yuriy1901) [reply src/components/Poem/Poem.js:169]: Yep. This code is already refactored on main branch. So when I wiil merge - I will take it frome there

File: src/App.js (hunk 1 above)
Comment (yuriy1901) [reply src/App.js]: You are right. Fixed

File: src/components/Poem/PoemBookmark/PoemBookmark.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+
+export default function PoemBookmark({ poem }) {
+    const dispatch = useDispatch();
+    const navigate = useNavigate();
+
+    const {isLoading, errorMessage, bookmarks} = useSelector(
+        (state) => state.bookmarks
+    )
+    const loginInfo = useSelector((state) => state.login);
```
[the remark this answers is not in the document]
Comment (yuriy1901) [reply src/components/Poem/PoemBookmark/PoemBookmark.js]: You are right. Fixed

File: src/pages/BookMarksPage/BookMarksPage.js (hunk 4 above)
Comment (yuriy1901) [reply src/pages/BookMarksPage/BookMarksPage.js]: You are right. Fixed

**Review by RudeAnna — APPROVED** [review]
Great job on bookmarks Yura

**Pull request merged**